### PR TITLE
Disaggregated heat load inputs and outputs + Process heat load

### PR DESCRIPTION
**Pull request by zolanaj**
### Please check if the PR fulfills these requirements
- [x] CHANGELOG.md is updated
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?
- Disaggregated heating loads are now available by heating load type: DomesticHotWaterLoad and SpaceHeatingLoad are now available separately
- New heating load ProcessHeatLoad included
- New heat-load-specific outputs for heating technologies added

### Does this PR introduce a breaking change?
(What changes might users need to make in their application due to this PR?)
None known, but the new I/O requires linkage to v. 0.46.1 of REopt.jl

### CODE REVIEW

**Comment by Bill-Becker** [comment]
@zolanaj just making sure this won't affect any existing behavior when using the default `can_serve_...` and that the outputs for HeatingLoad should all be unchanged as well. I'm making sure about this for the web tool interfacing with the API.

**Comment by zolanaj** [comment]
> @zolanaj just making sure this won't affect any existing behavior when using the default `can_serve_...` and that the outputs for HeatingLoad should all be unchanged as well. I'm making sure about this for the web tool interfacing with the API.

Confirmed, the defaults for [CHP](https://github.com/NREL/REopt_API/blob/44e40273908586036ac17b7ee009161312c730e9/reoptjl/models.py#L4108-L4124), [ExistingBoiler](https://github.com/NREL/REopt_API/blob/44e40273908586036ac17b7ee009161312c730e9/reoptjl/models.py#L4957-L4976), [HotThermalStorage](https://github.com/NREL/REopt_API/blob/44e40273908586036ac17b7ee009161312c730e9/reoptjl/models.py#L5824-L5841), [Boiler](https://github.com/NREL/REopt_API/blob/44e40273908586036ac17b7ee009161312c730e9/reoptjl/models.py#L5229-L5248), and [SteamTurbine](https://github.com/NREL/REopt_API/blob/44e40273908586036ac17b7ee009161312c730e9/reoptjl/models.py#L5562-L5580)  can all serve both DHW and space heating by default, which is consistent with the notion that all techs serve all heating loads.  The exception is [GHP](https://github.com/NREL/REopt_API/blob/44e40273908586036ac17b7ee009161312c730e9/reoptjl/models.py#L7111-L7123) which already had the field `can_serve_dhw` set to False as default.

HeatingLoad outputs for DHW and SpaceHeating are unchanged. There are new outputs within HeatingLoad related to process heat, but only if included (otherwise it's blank).  Will that work or should those be suppressed somehow?

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +26,19 @@ Classify the change according to the following categories:
     ##### Removed
     ### Patches
 
+## Develop 2024-05-09
```
Comment (adfarth) [inline CHANGELOG.md:29]: looks like retire_in_optimal is missing from this list

File: reoptjl/models.py [hunk 2]
```diff
@@ -6420,6 +6693,12 @@ class AbsorptionChillerInputs(BaseModel, models.Model):
         'hot_water'
     ))
 
+    HEATING_LOAD_INPUT = models.TextChoices('HEATING_LOAD_INPUT', (
+        'DomesitHotWater',
```
Comment (adfarth) [inline reoptjl/models.py]: fixed this spelling in https://github.com/NREL/REopt_API/pull/583/commits/abf56a1e8cf0c94e1e24797785f88290bf010b69

File: reoptjl/models.py [hunk 3]
```diff
@@ -6193,6 +6371,55 @@ def clean(self):
         if self.addressable_load_fraction == None:
             self.addressable_load_fraction = list([1.0]) # should not convert to timeseries, in case it is to be used with monthly_mmbtu or annual_mmbtu
 
+class ProcessHeatLoadInputs(BaseModel, models.Model):
+    # DHW
```
Comment (adfarth) [inline reoptjl/models.py:6375]: Is "#DHW" here intentional?

File: reoptjl/views.py [hunk 4]
```diff
@@ -14,7 +14,7 @@
     CoolingLoadOutputs, HeatingLoadOutputs, REoptjlMessageOutputs, HotThermalStorageInputs, HotThermalStorageOutputs,\
     ColdThermalStorageInputs, ColdThermalStorageOutputs, AbsorptionChillerInputs, AbsorptionChillerOutputs,\
     FinancialInputs, FinancialOutputs, UserUnlinkedRuns, BoilerInputs, BoilerOutputs, SteamTurbineInputs, \
-    SteamTurbineOutputs, GHPInputs, GHPOutputs
+    SteamTurbineOutputs, GHPInputs, GHPOutputs, ProcessHeatLoadInputs
```
Comment (adfarth) [inline reoptjl/views.py:17]: Why does only "DomesticHotWaterLoad" and "ProcessHeatLoad" not have "Outputs" included here?

File: julia_src/Manifest.toml [hunk 5]
```diff
@@ -917,9 +917,11 @@ uuid = "3fa0cd96-eef1-5676-8a61-b3b8758bbffb"
 
 [[deps.REopt]]
 deps = ["ArchGDAL", "CSV", "CoolProp", "DataFrames", "Dates", "DelimitedFiles", "HTTP", "JLD", "JSON", "JuMP", "LinDistFlow", "LinearAlgebra", "Logging", "MathOptInterface", "Requires", "Roots", "Statistics", "TestEnv"]
-git-tree-sha1 = "d2bbcbb8344f19ed87ee8cb6a196d4ed8415255b"
+git-tree-sha1 = "28bbe232da58b84194f3c1b4f0d60d577f24ba2c"
+repo-rev = "fix-ghp-storage-dispatch"
+repo-url = "https://github.com/NREL/REopt.jl.git"
 uuid = "d36ad4e8-d74a-4f7a-ace1-eaea049febf6"
-version = "0.45.0"
+version = "0.46.0"
```
Comment (adfarth) [inline julia_src/Manifest.toml:924]: Just a note to update this to the 46.1 eventually

**Review by adfarth — APPROVED** [review]
Approved with one very minor comment :)

**Comment by Bill-Becker** [comment]
@zolanaj thanks for confirming, that should work.

File: reoptjl/models.py (hunk 2 above)
Comment (zolanaj) [reply reoptjl/models.py]: Thanks for catching this and the changelog update!

File: reoptjl/models.py (hunk 3 above)
Comment (zolanaj) [reply reoptjl/models.py]: It is not!  Updated it here - https://github.com/NREL/REopt_API/commit/d7d6652c4dd3dcc0970c7305c035673fd374f744

File: julia_src/Manifest.toml (hunk 5 above)
Comment (zolanaj) [reply julia_src/Manifest.toml]: Thanks for the reminder! Addressed here: https://github.com/NREL/REopt_API/commit/ae8fceac0a30a8552d465260066a59d28f2edd00

**Pull request merged**